src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposableTests.cs

```diff
@@ -1166,6 +1166,49 @@ End Sub
         Throw New NotImplementedException()
     End Sub
 End Class
+");
+        }
+
+        [Fact]
+        [WorkItem(6151, "https://github.com/dotnet/roslyn-analyzers/issues/6151")]
+        public async Task CA1001CSharpDisposedCalledInStructAsync()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.IO;
+
+internal struct Invalid : IDisposable
```

Comment:
I would make it `public` to ensure it's always analyzed (we have some configuration to reduce scope).